packages/utils/src/store/encryption-codec.ts

```diff
@@ -0,0 +1,94 @@
+import { decode as base64Decode, encode as base64Encode } from "@stablelib/base64";
+import nacl from "tweetnacl";
+
+export interface EncryptionCodec {
+  encode(data: Uint8Array, userKey: Uint8Array): string;
+  decode(encoded: string, userKey: Uint8Array): Uint8Array;
+}
+
+/**
+ * Creates an encryption key directly from user ID
+ */
+export function createUserEncryptionKey(userId: string): Uint8Array {
+  // Convert user ID to bytes and pad/truncate to 32 bytes for NaCl
+  const encoder = new TextEncoder();
+  const userIdBytes = encoder.encode(userId);
+
+  // Create a 32-byte key from user ID
+  const key = new Uint8Array(32);
+
+  // If user ID is shorter than 32 bytes, repeat it
+  for (let i = 0; i < 32; i++) {
+    key[i] = userIdBytes[i % userIdBytes.length];
+  }
+
+  return key;
+}
+
+/**
+ * Encryption codec that encrypts data with user-specific keys
+ */
+export class UserEncryptionCodec implements EncryptionCodec {
+  private keyPair: nacl.BoxKeyPair;
+
+  constructor(userEncryptionKey: Uint8Array) {
+    // Derive a key pair from the user key for encryption
+    this.keyPair = nacl.box.keyPair.fromSecretKey(userEncryptionKey.slice(0, 32));
+  }
+
+  static createUserEncryptionCodec(userId: string): UserEncryptionCodec {
+    const userEncryptionKey = createUserEncryptionKey(userId);
+    return new UserEncryptionCodec(userEncryptionKey);
+  }
+
+  encode(data: Uint8Array): string {
+    try {
+      // Generate ephemeral key pair for this encryption
+      const ephemeralKeyPair = nacl.box.keyPair();
```

Comment:
How did you end up deciding on asymmetric encryption? `secretbox` with a constant nonce would work just fine for our use case.